library/js/oeUI/insurance/EditPolicyScreenController.js

```diff
@@ -0,0 +1,910 @@
+            // no template, nothing to do as the feature is turned off or we are not primary or secondary insurance
+            return;
+        }
+        let insuranceInfoContainer = document.getElementById('insurance-info-type-' + type);
+        if (!insuranceInfoContainer) {
+            throw new Error("Failed to find insurance info container for type: " + type);
+        }
+        let fieldDestination = insuranceInfoContainer.querySelector('.js-swap-insurance');
+        if (!fieldDestination) {
+            throw new Error("Failed to find swap insurance field for type: " + type);
+        }
+        let clone = document.importNode(template.content, true);
+        fieldDestination.parentNode.insertBefore(clone, fieldDestination.nextSibling);
+        let swapButton = insuranceInfoContainer.querySelector('.btn-insurance-swap[data-type="'+ type + '"]');
+        if (!swapButton) {
+            throw new Error("Failed to find swap button for type: " + type);
+        }
+        swapButton.classList.remove("d-none");
+        swapButton.addEventListener('click', (evt) => {
+            let targetType = evt.target.dataset['targetType'];
+            if (!targetType) {
+                throw new Error("Failed to find target type for swap button");
+            }
+            evt.target.disabled = true;
+            evt.target.querySelector('.wait').classList.remove('d-none');
+            this.__isSaving = true;
+            this.__policySaved = false;
+            this.__insurancePolicyService.swapInsurance(this.selectedInsurance, targetType)
+                .then(selectedInsurance => {
+                    evt.target.disabled = false;
+                    evt.target.querySelector('.wait').classList.add('d-none');
+                    // we can use our setup function to restart everything with the swapped policies
+                    this.setupNewPolicyEdit(selectedInsurance);
+                    // these fields get reset by the setupNewPolicyEdit function... so we need to undo it
+                    this.__isSaving = false;
+                    this.__policySaved = true;
+                    this.render();
+                })
+                .catch(error => {
+                    this.__isSaving = false;
+                    evt.target.disabled = false;
+                    evt.target.querySelector('.wait').classList.add('d-none');
+                    console.error(error);
+                    // if we couldn't swap due to validation errors... we need to display those errors
+                    if (error instanceof ValidationError) {
+                        this.__validationErrors = error.validationErrors;
+                        this.render(); // re-render the screen to update the display with the errors.
+                    } else {
+                        alert(window.top.xl("An error occurred while swapping the insurance policy.  Please try again or contact your system administrator."));
```

Comment:
only use single spacing in translated constants

Reply:
Fixed.